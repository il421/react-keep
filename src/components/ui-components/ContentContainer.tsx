import React, {CSSProperties} from "react";
import "../../styles/components/common/_content-conteiner.scss";

interface ContentContainerProps extends Partial<Pick<HTMLElement, "className">> {
  style?: CSSProperties
}

export class ContentContainer extends React.PureComponent<ContentContainerProps> {
  render() {
    return (
      <div
        className={`content-container ${ this.props.className }`}
        style={this.props.style}
      >{ this.props.children }
      </div>
    );
  }
}
