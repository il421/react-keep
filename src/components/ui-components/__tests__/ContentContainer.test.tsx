import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { ContentContainerProps, ContentContainer } from "../ContentContainer";

let props: ContentContainerProps,
  wrapper: ShallowWrapper<ContentContainerProps, {}>;

beforeEach(() => {
  props = {};
});

test("should render ContentContainer", () => {
  wrapper = shallow<ContentContainerProps>(<ContentContainer {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
