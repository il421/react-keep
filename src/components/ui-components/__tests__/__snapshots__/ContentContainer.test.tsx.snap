// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ContentContainer 1`] = `
<div
  className="content-container undefined"
/>
`;
